Sources/NIO/EventLoop.swift

```diff
@@ -658,7 +658,7 @@ private enum EventLoopLifecycleState {
 @usableFromInline
 internal final class SelectableEventLoop: EventLoop {
     private let selector: NIO.Selector<NIORegistration>
-    private let thread: NIOThread
+    internal let thread: NIOThread
```

Comment:
Instead of blowing up this abstraction, should we have `SelectableEventLoop` have a function that can be called that calls `join()`? Maybe a `destroy` function or similar?

Reply:
yeah, I'll probably to a refactor there. I'll try not to go nuts because there's a lot that could be refactored :P

Reply:
this is now done.